### ISSUE TITLE
Project file doesn't contain last version of SQL tab content (previously saved)

### ISSUE THREAD

**Issue by kanter** [sqlitebrowser/sqlitebrowser#2972]
Details for the issue
--------------------

#### What did you do?
I create a project and I open or create some SQL file and fill it with somes queries and I save the project. I modify again the SQL file and I save again the project. When DB browser crashed (yes it happens often), I reopen the project and : 
1 -  the open tabs with my previous SQL file and not linked to the real file on the harddrive anymore
2 - the content of the tab does not correspond to the last saved projet but seems to be an old version.

#### What did you expect to see?
When I save the projet, I would like to save uptodate tab content.

#### What did you see instead?
already explained higher

Sorry for my english langage, I'm french and not fluent in english

Useful extra information
-------------------------
> DB4S v3.12.2 [built for x86_64-little_endian-llp64] on Windows 10 (10.0) (winnt/10.0.18363) [x86_64]
> using SQLCipher Version 4.4.0 community (based on SQLite 3.34.1)
> and Qt 5.12.8

**Comment by chrisjlocke** [comment]
> the open tabs with my previous SQL file and not linked to the real file on the harddrive anymore

This was explained in your other issue you raised.

> the content of the tab does not correspond to the last saved projet

If you save the file, make edits, and DB4S crashes, then it won't have saved changes.  Or do you mean (for example) you save 'aaa' to the file, edit to 'bbb', save, edit to 'ccc', DB4S crashes, you open the file and it has reverted to 'aaa' ?  
As mentioned earlier, the project file doesn't store the link to the .sql file, so saving to the .sql file is more for 'information purposes only'.  In testing, I create a tab, save it, type in it, save the project, and the .sql file is empty.  It'll only get written to if I save to the SQL file again, not save the project.
So when you say 'the content of the tab', do you mean the file the tab points to (which it doesn't anymore) or the text in the tab?  In my testing, saving the project saved the tab text, and opening the project file kept this text.

**Comment by kanter** [comment]
Yes that’s it !! 
I save 'aaa' to the file, edit to 'bbb', save, edit to 'ccc', DB4S crashes, I open the file and it has reverted to 'aaa'

Envoyé de mon iPhone

**Comment by chrisjlocke** [comment]
Yes, so this makes sense.  As soon as you save the project, it no longer saves to the .SQL file - this is (I assume) by design.

**Comment by mgrojo** [comment]
@kanter: I've changed the behaviour. Now, the original state of each of the SQL "tabs" is restored. If the tab was
opened without a file, the content is saved in the project file itself (this hasn't changed); when the tab was opened from a file, the reference to the file is stored in the project file and the contents are restored from that disk file when loading the project. Could you give it a try using tomorrow's [nightly build](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Nightly-Builds) to confirm it's working fine for you?
